keylime/tenant.py

```diff
@@ -951,7 +976,7 @@ def main(argv=sys.argv):
     """
     parser = argparse.ArgumentParser(argv[0])
     parser.add_argument('-c', '--command', action='store', dest='command', default='add',
-                        help="valid commands are add,delete,update,status,list,reactivate,regdelete. defaults to add")
+                        help="valid commands are add,delete,smartdelete,update,status,list,reactivate,smartreactivate,regdelete. defaults to add")
```

Comment:
What are these new commands "smartdelete" and "smartreactivate" do that's different from regular "delete" and "reactivate"? It seems like they are smart enough to look up the agent first to determine it's verifier's ip/port combo, but why not have the regular delete/reactivate do that? Why do we need a separate command?

Reply:
Both `smartdelete` and `smartreactivate` require additional REST call to find the correct verifier instances` information in the case when it is unknown. In the case when the verifier's information is already available, adding this functionality to the existing `delete` and `reactivate` would have introduced unnecessary overhead.

Reply:
If the information is already known, couldn't that be used as the conditional to decide if the extra "smart" API calls are needed?

Reply:
In that case how would the tenant know, if the provided information is actually correct i.e. how would the tenant know if it should just trust the provided information or make the "smart" API calls?

Reply:
I'm still a bit fuzzy on why these extra smart APIs need to exist vs just enhance the existing APIs. Can you walk me through the use case and API calls (not just the individual calls, but the whole sequence that leads to them)?

Reply:
As discussed, the `smartdelete` and `smartreactivate` will be converted to the default `delete` and `reactivate`, and the original non-smart functionality will be available through a flag.